Reenable ruff

kiendang opened the pull request:
## Description
ruff pre-commit was misconfigured and as a result disabled for the whole codebase

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

kiendang committed:
Reenable ruff

exclude mongomock since we pasted it from elsewhere

kiendang committed:
Ruff autofix

kiendang committed:
Fix lint

kiendang commented:
@abyesilyurt Could you help fix the few remaining ones? Thanks!

```
packages/syft/tests/syft/service/sync/sync_resolve_single_test.py:90:16: B011 Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`
packages/syft/tests/syft/stores/kv_document_store_test.py:47:10: B017 `pytest.raises(Exception)` should be considered evil
packages/syft/tests/syft/stores/kv_document_store_test.py:82:10: B017 `pytest.raises(Exception)` should be considered evil
tests/integration/local/twin_api_sync_test.py:191:16: B011 Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`
```

kiendang commented on .pre-commit-config.yaml:
```diff
@@ -88,7 +88,7 @@ repos:
     hooks:
       - id: ruff
         args: [--fix, --exit-non-zero-on-fix, --show-fixes]
-        files: "^notebooks/(api|tutorials|admin)"
+        exclude: packages/syft/tests/mongomock
```
@yashgorana help me check pls. Thanks!

kiendang replied:
I think `files: "^notebooks/(api|tutorials|admin)"` was added wrongly to the `ruff` step instead of the `nbqa` one right above.

yashgorana replied:
mmm weird - had just made changes to clean up the regex in 4d1c349 but this line was introduced in 2b31864. unsure why but good that this was caught!

kiendang committed:
Update ruff version

tcp commented on packages/syft/src/syft/service/settings/settings.py:
```diff
@@ -1,5 +1,5 @@
 # stdlib
-from typing import Callable
+from collections.abc import Callable
```
Is this intended @kiendang?

kiendang replied:
yes `typing.Callable` is deprecated, replaced by `collections.abc.Callable` https://docs.python.org/3/library/typing.html#typing.Callable

kiendang committed:
Add path filter for nbqa

kiendang committed:
Fix lint B011

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

kiendang committed:
Fix lint B017

`pytest.raises(Exception)` should be considered evil

kiendang committed:
Merge branch 'dev' into reenable-ruff

kiendang committed:
Fix lint
